File: docs-examples/example-http-function-java/src/main/java/example/Book.java
package example;

import io.micronaut.core.annotation.Introspected;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@Introspected
public class Book {
    @NotBlank
    @NotNull
    private final String title;
    @Min(100)
    private final int pages;

    public Book(String title, int pages) {
        this.title = title;
        this.pages = pages;
    }

    public String getTitle() {
        return title;
    }

    public int getPages() {
        return pages;
    }
}

File: docs-examples/example-http-function-python/src/main/python/example/Book.py
from dataclasses import dataclass
from typing import Annotated

from jakarta.validation.constraints import Min, NotBlank, NotNull
from micronaut.serde.annotation import Serdeable


@Serdeable
@dataclass
class Book:
    title: Annotated[str, NotBlank, NotNull]
    pages: Annotated[int, Min(100)]
